new-log-viewer/src/components/modals/SettingsModal/ConfigDialog.tsx

```diff
@@ -0,0 +1,209 @@
+    });
+
+    if (null !== error) {
+        // eslint-disable-next-line no-warning-comments
+        // TODO: Show an error pop-up once NotificationProvider is implemented.
+        // eslint-disable-next-line no-alert
+        window.alert(error);
+    } else {
+        window.location.reload();
+    }
+};
+
+/**
+ * Renders a configuration settings form.
+ *
+ * @return
+ */
+const ConfigDialog = forwardRef<HTMLFormElement>((_, ref) => {
+    const {setMode, mode} = useColorScheme();
+
+    return (
+        <form
+            ref={ref}
+            tabIndex={-1}
+            onReset={handleConfigFormReset}
+            onSubmit={handleConfigFormSubmit}
+        >
+            <ModalDialog
+                maxWidth={"lg"}
+                size={"lg"}
+            >
+                <DialogTitle className={"settings-modal"}>
+                    <span
+                        style={{flexGrow: 1,
+                            fontSize: "2rem"}}
+                    >
+                        Settings
+                    </span>
+                    <ToggleButtonGroup
+                        size={"sm"}
+                        value={mode as string}
+                        onChange={(__, newValue) => {
+                            setMode(newValue as Mode);
+                        }}
+                    >
+                        <Button
+                            startDecorator={<LightMode/>}
+                            value={THEME_NAME.LIGHT}
+                        >
+                            Light
+                        </Button>
+                        <Button
+                            startDecorator={<SettingsBrightnessIcon/>}
+                            value={THEME_NAME.SYSTEM}
+                        >
+                            System
+                        </Button>
+                        <Button
+                            startDecorator={<DarkMode/>}
+                            value={THEME_NAME.DARK}
+                        >
+                            Dark
+                        </Button>
+                    </ToggleButtonGroup>
+                </DialogTitle>
+                <DialogContent>
+                    {CONFIG_FORM_FIELDS.map((field, index) => (
+                        <FormControl
+                            className={"config-form-control"}
+                            key={index}
+                        >
+                            <FormLabel>
+                                {field.label}
+                            </FormLabel>
+                            <Input
+                                slotProps={{
+                                    input: {
+                                        defaultValue: field.initialValue,
+                                        name: field.name,
+                                        size: 100,
+                                        type: field.type,
+                                    },
+                                }}/>
```

Comment:
We don't have to pass the input props via `slotProps`, right?
```suggestion
                            <Input
                                defaultValue={field.initialValue}
                                name={field.name}
                                type={field.type}/>
```